### PR TITLE
Test setup

### PR DESCRIPTION
**Pull request by walkah**
Setting up for unit & integration (via puppeteer) testing. 

This is a revival of #41 (which got way out of date)

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  presets: [
+    ['@babel/preset-env', { targets: { node: 'current' } }],
+    '@babel/preset-typescript'
+  ]
+}
```
Comment (icidasset) [inline babel.config.js:6]: Does jest need this to compile individual typescript files, and/or the tests?

File: src/common/base64.test.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @jest-environment jsdom
+ */
+import * as base64 from './base64'
+
+test('can decode an encoded string', () => {
+  const testString = 'ab/c-d=='
+  const encoded = base64.urlEncode(testString)
+  expect(base64.urlDecode(encoded)).toEqual(testString)
```
Comment (icidasset) [inline src/common/base64.test.ts:9]: Does it automatically import `test()` and `expect()`?

**Review by icidasset — APPROVED** [review]
Ship iiit 🙏 😄

File: babel.config.js (hunk 1 above)
Comment (walkah) [reply babel.config.js:6]: yep, exactly

File: src/common/base64.test.ts (hunk 2 above)
Comment (walkah) [reply src/common/base64.test.ts:9]: it does, which not everyone is a fan of (this is one of the things `ava` explicitly does differently)

**Pull request merged**